hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconScmTask.java

```diff
@@ -88,8 +94,12 @@ public boolean isRunning() {
   }
 
   protected void recordSingleRunCompletion() {
-    reconTaskStatusDao.update(new ReconTaskStatus(getTaskName(),
-        System.currentTimeMillis(), 0L));
+    taskStatusCounter.updateCounter(getTaskName(), true);
+    synchronized (this) {
+      reconTaskStatusRecord.setLastUpdatedTimestamp(System.currentTimeMillis());
+      reconTaskStatusRecord.setLastTaskRunStatus(1);
```

Comment:
But since we are using an INTEGER value, we do not have any indication of if the last task run was ever completed.
If the initialization is done using 0 then the result might give us false indication that the task had run even in the scenario where the task had not at all run.

So there should be another value with which we can initialize to indicate the task has not yet run (say in case Recon first comes up, and for some reason the task is not running even for the first time)

In the current code:
0 -> Equivalent to NULL, means no last task was run, we do not know status
1 -> Last task has passed
-1 -> Last task has failed

Reply:
As discussed we are using the exit code 0 for success and -1 for failure.
Addressed this in [90c1997](https://github.com/apache/ozone/pull/7517/commits/90c1997d1a1c2906198d2117542ba2fa10869a56)